src/components/bookmark-list-container.ts

```diff
@@ -288,60 +142,28 @@ export class BookmarkListContainer extends LitElement {
   };
 
   private handleSyncRequested = async () => {
-    if (this.containerState.isSyncing) return;
-
-    try {
-      const settings = await DatabaseService.getSettings();
-      if (settings) {
-        await SyncService.syncBookmarks(settings);
-        // Don't dispatch sync-requested event to avoid infinite loop
-        // The sync service itself will dispatch the necessary events
-      }
-    } catch (error) {
-      console.error('Failed to sync bookmarks:', error);
-    }
+    await this.syncController.requestSync();
   };
 
   private handleFaviconLoadRequested = async (bookmarkId: number, faviconUrl: string) => {
-    try {
-      const faviconDataUrl = await FaviconService.getFaviconForBookmark(bookmarkId, faviconUrl);
-      
-      // Update cache
-      const updatedCache = new Map(this.containerState.faviconCache);
-      updatedCache.set(bookmarkId, faviconDataUrl);
-      
-      this.containerState = {
-        ...this.containerState,
-        faviconCache: updatedCache,
-      };
-    } catch (error) {
-      console.error('Failed to load favicon:', error);
-    }
+    await this.faviconController.loadFavicon(bookmarkId, faviconUrl);
   };
 
   private handleVisibilityChanged = (visibleBookmarkIds: number[]) => {
-    // Load favicons for visible bookmarks
-    visibleBookmarkIds.forEach(bookmarkId => {
-      const bookmark = this.containerState.bookmarks.find(b => b.id === bookmarkId);
-      if (bookmark && bookmark.favicon_url && !this.containerState.faviconCache.has(bookmarkId)) {
-        this.handleFaviconLoadRequested(bookmarkId, bookmark.favicon_url);
-      }
-    });
+    this.faviconController.handleVisibilityChanged(visibleBookmarkIds, this.containerState.bookmarks);
   };
 
   override render() {
+    const syncState = this.syncController.getSyncState();
+    const faviconState = this.faviconController.getFaviconState();
+    
     return html`
       <bookmark-list
         .bookmarks=${this.containerState.bookmarks}
         .isLoading=${this.containerState.isLoading}
-        .syncState=${{
-          isSyncing: this.containerState.isSyncing,
-          syncProgress: this.containerState.syncProgress,
-          syncTotal: this.containerState.syncTotal,
-          syncedBookmarkIds: this.containerState.syncedBookmarkIds,
-        }}
+        .syncState=${syncState}
         .faviconState=${{
-          faviconCache: this.containerState.faviconCache,
+          faviconCache: faviconState.faviconCache,
           bookmarksWithAssets: this.containerState.bookmarksWithAssets,
```

Comment:
Why is `bookmarksWithAssets` part of `faviconState`? It's not related to favicons at all. It should be a separate property passed to BookmarkList.